lms/djangoapps/course_api/tests/test_api.py

```diff
@@ -0,0 +1,185 @@
+"""
+Test for course API
+"""
+
+from django.contrib.auth.models import AnonymousUser
+from django.http import Http404
+from django.test import RequestFactory
+from rest_framework.exceptions import PermissionDenied
+
+from opaque_keys.edx.keys import CourseKey
+from xmodule.modulestore.tests.django_utils import SharedModuleStoreTestCase, ModuleStoreTestCase
+
+from ..api import course_detail, list_courses
+from .mixins import CourseApiFactoryMixin
+
+
+class CourseApiTestMixin(CourseApiFactoryMixin):
+    """
+    Establish basic functionality for Course API tests
+    """
+
+    maxDiff = 5000  # long enough to show mismatched dicts
+
+    expected_course_data = {
+        'course_id': u'edX/toy/2012_Fall',
+        'name': u'Toy Course',
+        'number': u'toy',
+        'org': u'edX',
+        'short_description': u'A course about toys.',
+        'media': {
+            'image': {
+                'uri': u'/c4x/edX/toy/asset/just_a_test.jpg',
+                'name': u'Course Image',
+                'description': u'',
+                'type': u'logo',
+            }
+        },
+        'start': u'2015-07-17T12:00:00Z',
+        'start_type': u'timestamp',
+        'start_display': u'July 17, 2015',
+        'end': u'2015-09-19T18:00:00Z',
+        'enrollment_start': u'2015-06-15T00:00:00Z',
+        'enrollment_end': u'2015-07-15T00:00:00Z',
+        'blocks_url': '/api/courses/v1/blocks/?course_id=edX%2Ftoy%2F2012_Fall',
+    }
+
+    @classmethod
+    def setUpClass(cls):
+        super(CourseApiTestMixin, cls).setUpClass()
+        cls.request_factory = RequestFactory()
+
+
+class TestGetCourseDetail(CourseApiTestMixin, SharedModuleStoreTestCase):
+    """
+    Test course_detail api function
+    """
+    @classmethod
+    def setUpClass(cls):
+        super(TestGetCourseDetail, cls).setUpClass()
+        cls.course = cls.create_course()
+        cls.hidden_course = cls.create_course(course=u'hidden', visible_to_staff_only=True)
+        cls.honor_user = cls.create_user('honor', is_staff=False)
+        cls.staff_user = cls.create_user('staff', is_staff=True)
+
+    def _make_api_call(self, requesting_user, target_user, course_key):
+        """
+        Call the `course_detail` api endpoint to get information on the course
+        identified by `course_key`.
+        """
+        request = self.request_factory.get('/')
+        request.user = requesting_user
+        return course_detail(request.user, target_user, course_key, request)
+
+    def test_get_existing_course(self):
+        course_key = CourseKey.from_string(u'edX/toy/2012_Fall')
```

Comment:
Instead of hardcoding the course_key strings, get the course_key by calling `cls.course.id`.